src/main/java/com/hongliangjie/fugue/utils/RandomUtils.java

```diff
@@ -1,12 +1,66 @@
 package com.hongliangjie.fugue.utils;
 
+import java.util.Random;
 import java.util.concurrent.ThreadLocalRandom;
 
 /**
  * Created by liangjie on 3/3/16.
  */
 public class RandomUtils {
-    public static Double NativeRandom(){
-        return ThreadLocalRandom.current().nextDouble();
+
+    protected abstract class RandomGNR{
+        public abstract double nextDouble();
+        public abstract int nextInt(int upper);
+    }
+
+    protected class NativeRandom extends RandomGNR{
+        @Override
+        public double nextDouble() {
+            return ThreadLocalRandom.current().nextDouble();
+        }
+
+        @Override
+        public int nextInt(int upper) { return ThreadLocalRandom.current().nextInt(0, upper); }
+    }
+
+    protected class DeterministicRandom extends RandomGNR{
+        private Random r;
+
+        public DeterministicRandom(){
+            r = new Random(22);
+        }
+
+        @Override
+        public double nextDouble(){
+            return r.nextDouble();
+        }
+
+        @Override
+        public int nextInt(int upper) { return r.nextInt(upper); }
+
     }
+
+    RandomGNR r;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/hongliangjie/Fugue/file/2183992999/issues/source?bid=3162994&fileBranchId=3165099#l43)